ocs_ci/deployment/vmware.py

```diff
@@ -783,6 +783,112 @@ def destroy_scaleup_nodes(
         terraform_scale_up.destroy(scale_up_terraform_var)
 
 
+class VSPHEREIPI(VSPHEREBASE):
+    """
+    A class to handle vSphere IPI specific deployment
+    """
+
+    def __init__(self):
+        super(VSPHEREIPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super(VSPHEREIPI.OCPDeployment, self).__init__()
+            self.helper_node_details = load_auth_config()["baremetal"]
+            self.ipi_details = load_auth_config()["vmware_ipi"]
+
+        def deploy_prereq(self):
+            """
+            Overriding deploy_prereq from parent. Perform all necessary
+            prerequisites for VSPHEREIPI here.
+            """
+            super(VSPHEREIPI.OCPDeployment, self).deploy_prereq()
+
+        def create_config(self):
+            """
+            Creates the OCP deploy config for the Bare Metal
+            """
+            # Generate install-config from template
+            _templating = Templating()
+            ocp_install_template = (
+                f"install-config-{self.deployment_platform}-"
+                f"{self.deployment_type}.yaml.j2"
+            )
+            ocp_install_template_path = os.path.join(
+                "ocp-deployment", ocp_install_template
+            )
+            install_config_str = _templating.render_template(
+                ocp_install_template_path, config.ENV_DATA
+            )
+            install_config_obj = yaml.safe_load(install_config_str)
+            install_config_obj["pullSecret"] = self.get_pull_secret()
+            install_config_obj["sshKey"] = self.get_ssh_key()
+            install_config_obj["platform"]["vsphere"]["apiVIP"] = self.ipi_details.get('vmware_ipi_api_vip')
+            install_config_obj["platform"]["vsphere"]["ingressVIP"] = self.ipi_details.get('vmware_ipi_ingress_vip')
+            install_config_obj["metadata"]["name"] = self.ipi_details.get('vmware_ipi_default_cluster_name')
+            install_config_obj["baseDomain"] = self.ipi_details.get('vmware_ipi_default_base_domain')
+            install_config_str = yaml.safe_dump(install_config_obj)
+            install_config = os.path.join(self.cluster_path, "install-config.yaml")
+            install_config_backup = os.path.join(
+                self.cluster_path, "install-config.yaml.backup"
+            )
+            with open(install_config, "w") as f:
+                f.write(install_config_str)
+            with open(install_config_backup, "w") as f:
+                f.write(install_config_str)
+
+        def deploy(self, log_cli_level="DEBUG"):
+            """
+            Deployment specific to OCP cluster on this platform
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+            """
```

Comment:
empty lines are missing
```suggestion
            Deployment specific to OCP cluster on this platform
            
            Args:
                log_cli_level (str): openshift installer's log level
                    (default: "DEBUG")
                    
            """
```

Reply:
Done